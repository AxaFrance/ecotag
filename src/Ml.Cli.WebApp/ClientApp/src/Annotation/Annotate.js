import React, {useState} from "react";
import {Header, Name, Title} from "@axa-fr/react-toolkit-layout-header";
import logo from '@axa-fr/react-toolkit-core/dist/assets/logo-axa.svg';
import DatasetHandler from "./DatasetHandler";
import TableAnnotate from "./TableAnnotate";
import {QueryClient, QueryClientProvider} from "react-query";

const queryClient = new QueryClient();

const Annotate = ({MonacoEditor}) => {

    const [state, setState] = useState({
        fileName: "Annoter un dataset",
        items: []
    });

    return (
        <QueryClientProvider client={queryClient}>
            <Header>
                <Name
                    title="ML-CLI"
                    subtitle="Made by AXA"
                    img={logo}
                    alt="Logo AXA"
                />
            </Header>
            <Title
                title={state.fileName === "Annoter un dataset" ? state.fileName : `Fichier en cours de visualisation : ${state.fileName}`}/>
            <DatasetHandler state={state} setState={setState}/>
            {state.items.length > 0 &&
            <TableAnnotate state={state} MonacoEditor={MonacoEditor}/>
            }
        </QueryClientProvider>
    );
};

export default Annotate;
