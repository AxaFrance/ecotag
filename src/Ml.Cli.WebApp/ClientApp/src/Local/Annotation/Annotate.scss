.error-message{
  text-align: center;
  color: red;
  margin-bottom: 1em;
}

