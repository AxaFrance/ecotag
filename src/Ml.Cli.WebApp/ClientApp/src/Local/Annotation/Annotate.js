import React, {useState} from "react";
import {Header, Name} from "@axa-fr/react-toolkit-layout-header";
import logo from '@axa-fr/react-toolkit-core/dist/assets/logo-axa.svg';
import {QueryClient, QueryClientProvider} from "react-query";
import {ToastContainer} from "react-toastify";
import 'react-toastify/dist/ReactToastify.css';
import './Annotate.scss';
import Routes from "./Routes";

const queryClient = new QueryClient();

const Annotate = ({fetch}) => {

    const [state, setState] = useState({
        fileName: "Annotate a dataset",
        datasetLocation: "",
        items: [],
        annotationType: "JsonEditor",
        configuration: [{name: "Default", id: 0}],
        isFileInserted: false
    });

    return (
        <QueryClientProvider client={queryClient}>
            <Header>
                <Name
                    title="ML-CLI"
                    subtitle="Made by AXA"
                    img={logo}
                    alt="AXA Logo"
                />
            </Header>
            <Routes state={state} setState={setState} fetch={fetch}/>
            <ToastContainer/>
        </QueryClientProvider>
    );
};

export default React.memo(Annotate);
