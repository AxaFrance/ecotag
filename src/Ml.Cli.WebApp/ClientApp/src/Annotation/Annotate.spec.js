import React from "react";
import {fireEvent, render, waitFor} from "@testing-library/react";
import Annotate from "./Annotate";
import sleep from "../sleep";

const dataset = (`{"DatasetLocation": "C:\\\\location", "AnnotationType": "JsonEditor", "Configuration": \"[{\\\"name\\\": \\\"Recto\\\", \\\"id\\\": 0}, {\\\"name\\\": \\\"Verso\\\", \\\"id\\\": 1}]\", "Content": [{\"FileName\": \"{FileName}_pdf.json\",\"FileDirectory\": \"fileDirectoryValue\",\"ImageDirectory\": \"imageDirectoryValue\",\"Annotations\":{}}]}`);

const fetch = (status =200) => async (url, config) => {
    await sleep(1);
    console.log(url);
    return {
        status:status,
        json: async () => {
            await sleep(1);
            return ["C:\\\\imageLocation"];
        },
    };
};

describe("Check dataset handling", () => {
    test("Should insert dataset", async () => {
        const mockedMonacoEditor = () => (<div>This is a mocked Monaco Editor</div>);
        const {container, asFragment, getAllByText, getByAltText} = render(<Annotate MonacoEditor={mockedMonacoEditor} fetchFunction={fetch(200)}/>);

        const blob = new Blob([dataset], {type: "text/plain;charset=utf-8"});
        const fileSource = new File([blob], "dataSourceFile.json", {type: "application/json"});
        const fileInput = container.querySelector("input[type='file']");
        fireEvent.change(fileInput, {target: {files: [fileSource]}});

        await waitFor(() => expect(container.querySelector('.table-result')).not.toBeNull());
        await waitFor(() => expect(getAllByText(/Fichier en cours de visualisation : dataSourceFile.json/i)).not.toBeNull());
        await waitFor(() => expect(getByAltText(/file_image/i)).not.toBeNull());
        expect(asFragment()).toMatchSnapshot();
    });
});
