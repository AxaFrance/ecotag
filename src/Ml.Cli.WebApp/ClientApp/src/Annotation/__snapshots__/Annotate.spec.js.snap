// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Check dataset handling Should insert dataset and select ocr and rotation annotation types 1`] = `
<DocumentFragment>
  <div
    class="af-header"
  >
    <div
      class="container af-header__wrapper"
    >
      <header
        class="af-header__content"
        role="banner"
      >
        <div
          class="af-header__name"
        >
          <a
            class="af-logo"
            href="/"
          >
            <img
              alt="Logo AXA"
              class="af-logo__brand"
              src="logo-axa.svg"
            />
          </a>
          <h2
            class="af-header__title"
          >
            ML-CLI
            <span
              class="af-header__subtitle"
            >
              Made by AXA
            </span>
          </h2>
        </div>
      </header>
    </div>
  </div>
  <div
    class="af-title-bar af-title-bar--backhome"
  >
    <div
      class="container-fluid container af-title-bar__wrapper"
    >
      <a
        class="btn af-btn--circle"
        href="/"
        title="Retour à l'accueil"
      >
        <i
          class="glyphicon glyphicon-home"
        />
      </a>
      <h1
        class="af-title-bar__title"
      >
        Fichier en cours de visualisation : dataSourceFile.json
      </h1>
    </div>
  </div>
  <div
    class="file-loader__container"
  >
    <div
      class="file-loader__input"
    >
      <div
        class="af-form__file-input"
      >
        <div
          class="drop-box hidden-mobile"
          id="dataset_loader"
          tabindex="0"
        >
          <input
            accept="application/json"
            autocomplete="off"
            multiple=""
            name="placeDatasetFile"
            style="display: none;"
            tabindex="-1"
            type="file"
          />
          <div>
            Glissez/déposez vos fichiers
          </div>
        </div>
        <button
          class="af-btn af-btn--file af-btn--hasiconLeft"
          type="button"
        >
          <i
            class="glyphicon glyphicon-open"
          />
           Parcourir
        </button>
      </div>
    </div>
    <div
      class="library__container"
    >
      <p
        class="library__title"
      >
        Fichiers de test
      </p>
      <div
        class="library__files-list"
      >
        <div
          class="library__file"
        >
          <a
            download="/api/files/value=C%3A%5C%5CimageLocation"
            href=""
          >
            imageLocation
          </a>
          <span
            class="glyphicon glyphicon-play library__custom-glyphicon-play"
          />
        </div>
      </div>
    </div>
  </div>
  <div
    class="table-result"
  >
    <div
      class="table-result__header"
    >
      <div>
        <p>
          Résultat fichier : {FileName}_pdf.json
        </p>
      </div>
    </div>
    <div
      class="images-editor-container"
    >
      <div
        class="images-editor-container__images_column"
      >
        <div
          class="images-list"
        >
          <div
            class="images-list__item"
          >
            <div
              class="images-list__image-container"
            >
              <img
                alt="file_image"
                class="images-list__file_image"
                src="api/files/value=C%3A%5C%5CimageLocation"
              />
            </div>
          </div>
        </div>
      </div>
      <div
        class="editor-container"
      >
        <div>
          This is a mocked Monaco Editor
        </div>
        <div
          class="editor-container__errors-container"
        >
          <div
            class="editor-container__error-message"
          >
            <span>
              Impossible de parser l'item {FileName}_pdf.json. Veuillez vous assurer que le contenu de ce fichier est au format JSON.
            </span>
          </div>
        </div>
        <div
          class="editor-container__button"
        >
          <button
            class="btn af-btn"
          >
            Sauvegarder
          </button>
        </div>
      </div>
    </div>
  </div>
  <div
    class="af-paging"
  >
    <div
      class="af-paging__limit"
    >
      <div
        class="af-paging__limit"
      >
        <form
          class="af-paging__form"
        >
          <div
            class="af-form__group"
          >
            <div
              class="col col-sm-2 col-md-2 col-lg-2 col-xl-2"
            >
              <label
                class="af-form__group-label"
                for="paging"
              >
                Afficher
              </label>
            </div>
            <div
              class="col col-sm-10 col-md-10 col-lg-10 col-xl-10"
            >
              <div
                class="af-form__select"
              >
                <div
                  class="af-form__select-container"
                >
                  <select
                    class="af-form__input-select"
                    id="paging"
                  >
                    <option
                      value="5"
                    >
                      5
                    </option>
                    <option
                      value="10"
                    >
                      10
                    </option>
                    <option
                      value="25"
                    >
                      25
                    </option>
                    <option
                      value="50"
                    >
                      50
                    </option>
                    <option
                      value="100"
                    >
                      100
                    </option>
                  </select>
                  <span
                    class="glyphicon glyphicon-menu-down"
                  />
                </div>
                <span
                  class="af-form__input-cmplt"
                >
                  éléments
                </span>
              </div>
            </div>
          </div>
        </form>
      </div>
    </div>
    <div
      class="af-paging__pager"
    >
      <nav
        class="af-pager"
      >
        <ul
          class="af-pager__pagination"
        >
          <li
            class="af-pager__item af-pager__item--disabled"
          >
            <span
              class="af-pager__item-nolink"
            >
              « Précédent
            </span>
          </li>
          <li
            class="af-pager__item af-pager__item--active"
          >
            <a
              class="af-pager__item-link"
              href="#"
            >
              <span>
                1
              </span>
            </a>
          </li>
          <li
            class="af-pager__item af-pager__item--disabled"
          >
            <span
              class="af-pager__item-nolink"
            >
              Suivant »
            </span>
          </li>
        </ul>
      </nav>
    </div>
  </div>
</DocumentFragment>
`;
